lib/pharos/phases/validate_host.rb

```diff
@@ -46,6 +48,11 @@ def validate_unique_hostnames
         raise Pharos::InvalidHostError, "Duplicate hostname #{@host.hostname} for hosts #{duplicates.map(&:address).join(',')}"
       end
 
+      def validate_localhost_resolve
+        localhost_ip = @ssh.exec!("ping -c 1 localhost | grep -Eo -m 1 '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}';")
+        raise Pharos::InvalidHostError, "Localhost ip does not look right: #{localhost_ip}" unless localhost_ip.start_with?('127.')
```

Comment:
Could be `IPAddr.new(localhost_ip).loopback?` (which needs `using Pharos::CoreExt::IPAddrLoopback if RUBY_VERSION < '2.5.0'`)